### DISCUSSION TITLE
a potential future Stable Diffusion implementation in ggml?

### DISCUSSION THREAD

**Discussion by ClashSAN** [ggml-org/ggml#303] [General]
I found this project: https://pixlab.io/tiny-dream 
"Stable Diffusion implementation written in C++ with main focus on CPU performance and memory footprint."

On their roadmap:
"Move the Tensor library from NCNN to a non bloated one such as [SOD](https://sod.pixlab.io/) or [GGML](https://github.com/ggerganov/ggml) with focus on CPU performance."

This would be very exciting to see, The spirit of [inference at the edge](https://github.com/ggerganov/llama.cpp/discussions/205) could allow users to experience incredible GPU acceleration speeds, as well as get stable diffusion running well on low ram: android, SBCs, web-browsers, and regular laptops.

Just wanted to inform:

 "Tiny-Dream is scheduled to be released on June/July 2023." 

@ggerganov maybe it's a good path for ggml to contact the developers at this time?

**Comment by ggerganov** [comment]
Very cool. I can't commit at the moment with a SD implementation as there are too many things I'm already trying to juggle, but I'm sure there will be people that will help out with the implementation, based on all the cool `ggml`-based projects that are already popping around
